### PR TITLE
Plugin run seq

### PR DESCRIPTION
**Pull request by acavit**

### CODE REVIEW

File: plugins/EpadDB.js [hunk 1]
```diff
@@ -2209,36 +2339,129 @@ async function epaddb(fastify, options, done) {
 
             const dock = new DockerService(fs, fastify, path);
             const containerName = `epadplugin_${pluginObj.id}`;
-            dock
+            // eslint-disable-next-line no-await-in-loop
+            await dock
```
Comment (emelalkim) [inline plugins/EpadDB.js:2343]: you cannot use both await and then. you should convert it to await structure (remove then and put an inner try catch for error handling)

File: plugins/EpadDB.js [hunk 2]
```diff
@@ -2209,36 +2339,129 @@ async function epaddb(fastify, options, done) {
...
+            }
+            await fastify.runPluginsQueueInternal(seqresult, request);
+          }
+        });
+    } catch (err) {
+      fastify.log.error(`runPluginsQueue error : ${err}`);
+    }
+  });
+
+  fastify.decorate('getNextPluginInSubQueue', async (paramQid, request) => {
```
Comment (emelalkim) [inline plugins/EpadDB.js:2407]: this doesn't return a plugin id or plugin info. I'm guessing it triggers plugin. please put a comment on what it does

File: plugins/EpadDB.js (hunk 1 above)
Comment (acavit) [reply plugins/EpadDB.js]: switched from then to await

File: plugins/EpadDB.js [hunk 3]
```diff
@@ -2334,23 +2335,19 @@ async function epaddb(fastify, options, done) {
...
             fastify.log.info('not a real error. Container has not found so we can create new one');
-            if (sequence) {
-              nonseqresult.push(pluginObj);
-              seqresult.push(pluginObj);
-            } else {
-              throw new Error('404');
+            nonseqresult.push(pluginObj);
+            seqresult.push(pluginObj);
+            if (!sequence) {
+              fastify.runPluginsQueueInternal(nonseqresult, request);
```
Comment (emelalkim) [inline plugins/EpadDB.js:2348]: don't you need await for runPluginsQueueInternal

File: plugins/EpadDB.js [hunk 4]
```diff
@@ -2303,7 +2302,9 @@ async function epaddb(fastify, options, done) {
     if (typeof request.body.sequence !== 'undefined') {
       sequence = request.body.sequence;
     }
-    const allStatus = ['added', 'ended', 'error', 'running'];
+
+    // const allStatus = ['added', 'ended', 'error', 'running', 'inqueue'];
+    const allStatus = ['added', 'ended', 'error'];
```
Comment (emelalkim) [inline plugins/EpadDB.js:2307]: did you remove inqueue?

**Review by emelalkim — APPROVED** [review]
just check my comments and merge when you are done

File: plugins/EpadDB.js (hunk 3 above)
Comment (acavit) [reply plugins/EpadDB.js:2348]: no we don't need await because this section runs in parallel

File: plugins/EpadDB.js (hunk 4 above)
Comment (acavit) [reply plugins/EpadDB.js:2307]: yes inqueue and running states removed to avoid to resend the plugin ids which are already running or inqueue.

**Pull request merged**